app/com/lucidchart/open/cashy/controllers/UploadController.scala

```diff
@@ -212,17 +195,17 @@ class UploadController extends AppController with ExtensionsConfig with UploadFe
     Action(parse.multipartFormData(FileHandler.handleFilePartAsByteArray)) { implicit request =>
       val response = uploadForm.bindFromRequest.fold(
         formWithErrors => formWithErrors.errors,
-        data => List.empty
+        data => {
+          Try {
+            AssetDataHelper.getData(request.body.file("assetFile"), data.assetURL)
```

Comment:
So, we essentially have to download each asset twice? Once to validate, and once to actually upload? I mean, the performance isn't a huge deal, since there aren't tons of concurrent cashy users, but is there any way we don't end up doing this?